testing/unittests/esdltests.cpp

```diff
@@ -1984,6 +1987,34 @@ constexpr const char * result = R"!!(<soap:Envelope xmlns:soap="http://schemas.x
       }
     }
 
+    void testParamEx()
+    {
+      static constexpr const char* input = R"!!!(
+        <root>
+          <Request/>
+        </root>
+      )!!!";
+      static constexpr const char* script = R"!!!(
+        <es:CustomRequestTransform xmlns:es="urn:hpcc:esdl:script" target="Request">
+          <es:param name="espUserStatusString" select="'defaulted'"/>
+          <es:assert test="$espUserStatusString != 'defaulted'" code="'1001'" message="'did not find defined input'"/>
+          <es:source xpath="Request">
+            <es:param name="bar" select="'defaulted'"/>
+            <es:assert test="$bar = 'defaulted'" code="'1002'" message="'did not assign explicit default value'"/>
+          </es:source>
+          <es:source xpath="Request">
+            <es:param name="baz"/>
+            <es:assert test="string-length($baz) = 0" code="'1003'" message="'did not assign implied default value'"/>
```

Comment:
Isn't the implied default value a string of zero length?

Reply:
Yes. Line 2015 declares 1004 to be the expected failure code, so this is not expected to fail and a failure would cause test case failure.